src/compiler/crystal/syntax/parser.cr

```diff
@@ -3702,14 +3704,19 @@ module Crystal
 
     def parse_visibility_modifier(modifier)
       doc = @token.doc
-
       next_token_skip_space
-      exp = parse_op_assign
 
-      modifier = VisibilityModifier.new(modifier, exp)
-      modifier.doc = doc
-      exp.doc = doc
-      modifier
+      if @token.type == :NEWLINE
```

Comment:
@asterite - A few questions:

1. To produce a `VisibilityModifier` with a nil expression, do you mean something like `VisibilityModifier.new(modifier, Nop.new)`?

2. To move the logic to the `TypeVisitor` (which I'm assuming is in the `visit(node : VisibilityModifier)` method in `type_inference.cr`), how would I access the subsequent `Def` nodes if the expression is a `Nop`?